### PR TITLE
User related db

### PR DESCRIPTION
**Pull request by sebastianibam002**
Two missing methods

### CODE REVIEW

File: src/main/java/controllers/MySQLController.java [hunk 1]
```diff
@@ -60,6 +60,110 @@ public void editPost(Post newPost) {
...
+    /**
+     * Given an user objects changes all the attributes
+     * such that match the ones given in the parameter
+     * except the id
+     * @param user user object that stores the id to
+     *             update the values
+     */
+    public boolean updateUser(User user){
+        try {
+            String query = "UPDATE FROM `user_info` SET `username`= ?, `password`=?, `bio`=? " +
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:72]: Please look at all the attributes of the User object and update the necessary tables in the database. For example, a User has likeHistory. We should be updating the "likes" table in the database in this method, ideally with a helper method. Please start this work ASAP as there is a lot to be done here. I think this should be really similar to addNewUser, if you want to reuse some of the code/helpers from there if it is possible

File: src/main/java/controllers/MySQLController.java [hunk 2]
```diff
@@ -60,6 +60,110 @@ public void editPost(Post newPost) {
...
+        this.deletingFromUser(user, "follows");
+    }
+
+    /**
+     * Given a correct tableName this method deletes the rows where
+     * the user_id is equal to the one given
+     * @param user: object user where we get the user_id
+     * @param tableName: the name of the table
+     */
+    private void deletingFromUser(User user, String tableName){
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:154]: Great idea using this method. Could we maybe rename the method to be more descriptive? A good name might be deleteUserDataFromTable

File: src/main/java/controllers/MySQLController.java [hunk 3]
```diff
@@ -60,6 +60,110 @@ public void editPost(Post newPost) {
...
+            String query = "DELETE FROM `user_info` WHERE `user_id`=?";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, user.getId());
+            preparedStatement.execute();
+
+        } catch (Exception e){
+            e.printStackTrace();
+            return false;
+        }
+        this.deletePostsFromUser(user);
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:107]: If these methods are one-liners that just call deletingFromUser, I think we could get rid of these and just call deletingFromUser directly.

**Review by ericjest — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/MySQLController.java (hunk 1 above)
Comment (sebastianibam002) [reply src/main/java/controllers/MySQLController.java:72]: Is it possible that inside users there could be a method that compares two users?

Comment (ericjest) [reply src/main/java/controllers/MySQLController.java:72]: What do you mean by "inside users"?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: src/main/java/controllers/MySQLController.java [hunk 4]
```diff
@@ -61,6 +61,180 @@ public void editPost(Post newPost) {
         }
     }
 
+    private User getUser(String userId){
+        User[] users = this.getAllUsers();
+        for(User user: users){
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:66]: space before ( for style consistency

**Review by shawnpl7 — APPROVED** [review]

File: src/main/java/controllers/MySQLController.java [hunk 5]
```diff
@@ -61,6 +61,180 @@ public void editPost(Post newPost) {
...
+            if (user.getId().equals(userId)) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    private boolean userHasPostsDB(User user){
+        try{
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:75]: space after try

File: src/main/java/controllers/MySQLController.java [hunk 6]
```diff
@@ -61,6 +61,180 @@ public void editPost(Post newPost) {
...
+        User[] users = this.getAllUsers();
+        for(User user: users){
+            if (user.getId().equals(userId)) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    private boolean userHasPostsDB(User user){
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:74]: space after user)

File: src/main/java/controllers/MySQLController.java [hunk 7]
```diff
@@ -71,6 +71,27 @@ private User getUser(String userId){
...
+        // first I delete the posts from the user
+        this.deletePostsFromUser(user);
+        // add the new ones
+        for (Post post: newUser.getPosts()){
+            this.addNewPost(new Post(user.getId(), post.getTime(), post.getRecipe(),
+                    post.getCategory(), post.getId()));
+        }
+
+    }
+
```
[the remark this answers is not in the document]
Comment (ericjest) [reply src/main/java/controllers/MySQLController.java:105]: Remove these new lines

**Review by ericjest — CHANGES_REQUESTED** [review]

**Review by ericjest — APPROVED** [review]

File: src/main/java/controllers/MySQLController.java [hunk 8]
```diff
@@ -54,6 +54,174 @@ public void editPost(Post newPost) {
...
+    /**
+     * Given a correct tableName this method deletes the rows where
+     * the user_id is equal to the one given
+     * @param user: object user where we get the user_id
+     * @param tableName: the name of the table
+     */
+    private void deleteUserDataFromTable(User user, String tableName){
+        try {
+            String query = "DELETE FROM `" + tableName + "` WHERE `user_id`=?";
```
Comment (ericjest) [inline src/main/java/controllers/MySQLController.java:213]: Change this ASAP so we call this code (line 212 to line 220) for each tableName. We cannot pass in a tableName dynamically.

**Review by ericjest — CHANGES_REQUESTED** [review]

**Review by ericjest — APPROVED** [review]

**Pull request merged**